### PR TITLE
AR pet show case

### PR DESCRIPTION
**Pull request by ragner**
* Play with pet in AR environment
* Includes physics simulation
* Android views
* BVH animations

GearVRf-DCO-1.0-Signed-off-by: Ragner Magalhaes <ragner.n@samsung.com>

### CODE REVIEW

**Comment by ragner** [comment]
this PR depends on https://github.com/Samsung/GearVRf/pull/2016

Dear @NolaDonato,
we are facing a issue with the collada file exported by @spodila1.

Sometimes the head of the pet is not rendering during the animations.

Could you please help us with this issue?

**Comment by ragner** [comment]
You need configure a API key to test the share functionality.  

https://developers.google.com/ar/develop/java/cloud-anchors/quickstart-android

File: common.gradle [hunk 1]
```diff
@@ -1,4 +1,7 @@
-project.ext.hasBackend = false
+if (project.hasProperty("hasBackend"))
```
Comment (liaxim) [inline common.gradle]: What is the purpose of this change? Why is it necessary?

**Comment by liaxim** [comment]
For me a strange plane was detected. Like floating in space and through an opened door. The pet face and the whole head get culled when in "bone" mode and the pet gets close to the camera.

File: common.gradle (hunk 1 above)
Comment (ragner) [reply common.gradle]: @liaxim This demo is for monoscopic mode only. How could we configure our project to build with monoscopic backend only?

Our try was done at commit 9ab5547d165d69d6e70a6978314fe0daef5bb530

**Comment by ragner** [comment]
@liaxim About the "pet getting close to the camera" we are fixing this 
but about the "whole head get culled when in bone mode" we don't know
why this is happening. Could it be a Framework's  issue?

**Comment by ragner** [comment]
Hi @NolaDonato,
we was using GVRBoxCollider to the Planes but it was not working properly. 
The Drag and GoToTap(Pet goes to tapped position in the plane) were not working because of this.

At commit 7b9c9a57c976a5d453aca47e5c4f9c0f59f26b21 we exchanged from Box to GVRMeshCollider.  GVRMeshCollider works as expected.

**Comment by ragner** [comment]
Hi @NolaDonato,
I am trying implement the animation of the pet grabbing the "bone toy" on the plane. 

I would attache the "bone toy" to a mouth's bone to have the animation of the pet grabbing the "bone toy" but it looks not working.

Do you have any tip how could I implement this?

**Comment by ragner** [comment]
Knowing issues:

1) pet’s head getting chopped off. (we need help with this)
2) Bone toy not attached to pet's mouth at the animation of taking the bone toy from the floor. (we need help with this)
3) Deadlock at our message service on share mode.(@acosta and @andrenaveca could you handle this?)
4) Guest not recovering after share mode has been finished. (@dsazulay  could you handle this?)

**Comment by liaxim** [comment]
Should we consider merging? What about moving this to sxrsdk-demos?

**Comment by liaxim** [comment]
Should we consider merging? What about moving this to sxrsdk-demos?

**Comment by liaxim** [comment]
Should we consider merging? What about moving this to sxrsdk-demos?

**Comment by ragner** [comment]
@liaxim You can close this PR.

We have moved to sxrsdk-demo. We will do a new PR to sxrsdk soon.
